pkg/controllers/allocation/controller.go

```diff
@@ -75,6 +78,11 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	// 1. Fetch provisioner
 	persistedProvisioner, provisionerWithDefaults, err := c.retrieveProvisionerFrom(ctx, req)
 	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
I think we can just do something like 
```
provisioner, err := c.provisionerFor(ctx, req)
```
since we no longer mutate or anything. Should clean up your variable names.